lib/ecmascript.ts

```diff
@@ -3850,397 +4063,366 @@ export const ES = ObjectAssign({}, ES2020, {
-      case 'object':
-      case 'number':
-      case 'symbol':
-      default:
-        throw new TypeError(`cannot convert ${typeof arg} to bigint`);
-      case 'string':
-        if (!prim.match(/^\s*(?:[+-]?\d+\s*)?$/)) {
-          throw new SyntaxError('invalid BigInt syntax');
-        }
-      // eslint: no-fallthrough: false
-      case 'bigint':
-        try {
-          return bigInt(prim as bigint);
-        } catch (e) {
-          if (e instanceof Error && e.message.startsWith('Invalid integer')) throw new SyntaxError(e.message);
-          throw e;
-        }
-      case 'boolean':
-        if (prim) {
-          return bigInt(1);
-        } else {
-          return bigInt.zero;
-        }
-    }
-  },
-
-  // Note: This method returns values with bogus nanoseconds based on the previous iteration's
-  // milliseconds. That way there is a guarantee that the full nanoseconds are always going to be
-  // increasing at least and that the microsecond and nanosecond fields are likely to be non-zero.
-  SystemUTCEpochNanoSeconds: (() => {
-    let ns = Date.now() % 1e6;
-    return () => {
-      const ms = Date.now();
-      const result = bigInt(ms).multiply(1e6).plus(ns);
-      ns = ms % 1e6;
-      return bigInt.min(NS_MAX, bigInt.max(NS_MIN, result));
-    };
-  })(),
-  SystemTimeZone: () => {
-    const fmt = new IntlDateTimeFormat('en-us');
-    const TemporalTimeZone = GetIntrinsic('%Temporal.TimeZone%');
-    return new TemporalTimeZone(ES.TemporalTimeZoneFromString(fmt.resolvedOptions().timeZone));
-  },
-  ComparisonResult: (value) => (value < 0 ? -1 : value > 0 ? 1 : value),
-  GetOptionsObject: (options) => {
-    if (options === undefined) return ObjectCreate(null);
-    if (ES.Type(options) === 'Object') return options;
-    throw new TypeError(
-      `Options parameter must be an object, not ${options === null ? 'null' : `a ${typeof options}`}`
-    );
-  },
-  GetOption: (options, property, allowedValues, fallback) => {
-    let value = options[property];
-    if (value !== undefined) {
-      value = ES.ToString(value);
-      if (!allowedValues.includes(value)) {
-        throw new RangeError(`${property} must be one of ${allowedValues.join(', ')}, not ${value}`);
-      }
-      return value;
-    }
-    return fallback;
-  },
-  GetNumberOption: (options, property, minimum, maximum, fallback) => {
-    let value = options[property];
-    if (value === undefined) return fallback;
-    value = ES.ToNumber(value);
-    if (NumberIsNaN(value) || value < minimum || value > maximum) {
-      throw new RangeError(`${property} must be between ${minimum} and ${maximum}, not ${value}`);
+// Note: This method returns values with bogus nanoseconds based on the previous iteration's
+// milliseconds. That way there is a guarantee that the full nanoseconds are always going to be
+// increasing at least and that the microsecond and nanosecond fields are likely to be non-zero.
+export const SystemUTCEpochNanoSeconds = (() => {
```

Comment:
This code is weird enough (took me a few times reading it before I understood what it was doing and why it was a const instead of an exported function) that it probably deserves an explicit type annotation on `SystemUTCEpochNanoSeconds`.

Reply:
Done.